rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -227,4 +251,98 @@ contract RBTCWrapperProxy{
         }        
     }
 
+    function addLiquidityToV1(
+        address _liquidityPoolConverterAddress,
+        IERC20Token[] memory _reserveTokens, 
+        uint256[] memory _reserveAmounts, 
+        uint256 _minReturn
+    )  
+        public
+        payable
+        checkAddress(_liquidityPoolConverterAddress)
+        returns(uint256) 
+    {
+        uint256 lengthOfTokens = _reserveTokens.length;
+
+        require(_reserveAmounts.length == lengthOfTokens.add(1), "The length of array is wrong");
+        require(_reserveAmounts[lengthOfTokens] == msg.value, "The provided amount of RBTC should be identical to msg.value");
+
+        bool successOfTransfer;
+        bool successOfApprove;
+        
+        for (uint256 i = 0; i < lengthOfTokens; i++) {
+            IERC20Token reserveToken = _reserveTokens[i];
+            uint256 reserveAmount = _reserveAmounts[i];
+            successOfTransfer = IERC20Token(reserveToken).transferFrom(msg.sender, address(this), reserveAmount);
+            require(successOfTransfer, "Failed to transfer reserve token from user");
+            successOfApprove = IERC20Token(reserveToken).approve(_liquidityPoolConverterAddress, reserveAmount);
+            require(successOfApprove, "Failed to approve converter to transfer reserve token");
+        }
+
+        (bool successOfDeposit, ) = wrbtcTokenAddress.call.value(_reserveAmounts[lengthOfTokens])(abi.encodeWithSignature("deposit()"));
+        require(successOfDeposit, "Failed to deposit RBTC to get WRBTC");
+        _reserveTokens[lengthOfTokens] = IERC20Token(wrbtcTokenAddress);
+        successOfApprove = IWrbtcERC20(wrbtcTokenAddress).approve(_liquidityPoolConverterAddress, _reserveAmounts[lengthOfTokens]);
+        require(successOfApprove, "Failed to approve converter to transfer WRBTC");
+
+        ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
+        ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+        uint256 poolTokenAmountBefore = _poolToken.balanceOf(address(this));
+        _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
```

Comment:
`LiquidityPoolV1Converter.addLiquidity` usually only partially uses `_reserveAmounts` - there are some leftovers due to trading and reserves fluctuations. It seems they will be stuck in this contract

Reply:
Code has been updated. And actually for now, we just wanna support `(W)RBTC <-> SOV`

Reply:
As for now, `addLiquidityToV1` neither fully utilizes `_reserveAmounts` nor sends the leftovers back, e.g.:
1. User checks WRBTC/SOV pool balance and decides to deposit `A` WRBTC and `B` SOV tokens (current price). The User sends a transaction with `_reserveAmounts = [A,B]`
2. Another User performs a trade and WRBTC/SOV price changes, e.g. SOV is more expensive now.
3. The deposit transaction is mined, and `LiquidityPoolV1Converter.addLiquidity` utilizes `[A, B-eps]` tokens.
4. `eps` SOV tokens are stuck on `RBTCWrapperProxy`

Reply:
@eMarchenko is completely right. the v1 pool will very often not utilize everything the user is providing. therefore, you need to send the remainder back.

Reply:
i wrote an explanation of how the v1 pool is treating transfers in the incorrect ratio on the sorcery channel yesterday.